### PR TITLE
mrc-4298 skeleton app 

### PR DESCRIPTION
**Pull request by hillalex**
Adds package skeleton, using Hatch, with some basic cli arg parsing to bootstrap it. See README for details on testing, building etc using Hatch.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    "Q",
+    "RUF",
+    "S",
+    "T",
+    "TID",
+    "UP",
+    "W",
+    "YTT",
+]
+ignore = [
```
Comment (richfitz) [inline pyproject.toml:125]: where did this set come from?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install hatch
+      - name: Lint
+        run: |
+          hatch run lint:style
+      - name: Test
+        run: |
+          hatch run cov
```
Comment (richfitz) [inline .github/workflows/test.yml:34]: can you wire this up to hit codecov.io or similar?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,95 @@
...
+To test the installation, first run Python in a container:
+
+```
+docker run --rm -it --entrypoint bash python
+```
+
+Then:
+
+```
+pip install --index-url https://test.pypi.org/simple packit-deploy --extra-index-url https://pypi.org/simple
```
Comment (richfitz) [inline README.md:86]: with some luck hatch will render this unnecessary as hopefully we can no longer publish non-functional packages by accident?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,95 @@
...
+```
+
+Now you should be able to run `packit` from the command line and see the usage instructions.
```
Comment (richfitz) [inline README.md:89]: The version on test.pypi is further along from the look of it?

**Review by richfitz — CHANGES_REQUESTED** [review]
LGTM, just the coverage to wire up and some assorted comments

File: README.md (hunk 4 above)
Comment (hillalex) [reply README.md:89]: yeah, I have a working version of the tool on there (code on further branches)

File: src/packit_deploy/cli.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
+"""Usage:
+  packit start <path> [--extra=PATH] [--option=OPTION]... [--pull]
+  packit status <path>
+  packit stop <path> [--volumes] [--network] [--kill] [--force]
```
[the remark this answers is not in the document]
Comment (hillalex) [reply src/packit_deploy/cli.py:4]: As in, version of this tool? Nice idea yeah

File: pyproject.toml (hunk 1 above)
Comment (hillalex) [reply pyproject.toml:125]: this was how hatch initialised the project

File: README.md (hunk 3 above)
Comment (richfitz) [reply README.md:86]: yeah, this is a good check - that was just a general whinge about the times this has burnt us!

**Review by richfitz — CHANGES_REQUESTED** [review]
The coverage upload is silently failing:

```Run codecov/codecov-action@v3
==> linux OS detected
https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
==> SHASUM file signed by key id 806bb28aed779869
==> Uploader SHASUM verified (02b055414cc124203ab061ffc45f5497ca39ede5e63a06717a6981c32d7b425f  codecov)
==> Running version latest
==> Running version v0.5.0
/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov -n  -Q github-action-3.1.4 -C 802e5ae52ba0ac7b51facfd2dc5b1436e90[10](https://github.com/mrc-ide/packit-deploy/actions/runs/5187098403/jobs/9349042993?pr=1#step:7:11)5ae
[2023-06-06T09:50:00.756Z] ['info'] 
     _____          _
    / ____|        | |
   | |     ___   __| | ___  ___ _____   __
   | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
   | |___| (_) | (_| |  __/ (_| (_) \ V /
    \_____\___/ \__,_|\___|\___\___/ \_/

  Codecov report uploader 0.5.0
[[20](https://github.com/mrc-ide/packit-deploy/actions/runs/5187098403/jobs/9349042993?pr=1#step:7:21)[23](https://github.com/mrc-ide/packit-deploy/actions/runs/5187098403/jobs/9349042993?pr=1#step:7:24)-06-06T09:50:00.764Z] ['info'] => Project root located at: /home/runner/work/packit-deploy/packit-deploy
[2023-06-06T09:50:00.766Z] ['info'] -> No token specified or token is empty
[2023-06-06T09:50:00.776Z] ['info'] Searching for coverage files...
[2023-06-06T09:50:00.883Z] ['info'] Warning: Some files located via search were excluded from upload.
[2023-06-06T09:50:00.883Z] ['info'] If Codecov did not locate your files, please review https://docs.codecov.com/docs/supported-report-formats
[2023-06-06T09:50:00.883Z] ['error'] There was an error running the uploader: No coverage files located, please try use `-f`, or change the project root with `-R`
```

**Review by richfitz — APPROVED** [review]

**Pull request merged**